x/ccv/provider/keeper/distribution.go

```diff
@@ -57,32 +68,149 @@ func (k Keeper) GetAllConsumerRewardDenoms(ctx sdk.Context) (consumerRewardDenom
 	return consumerRewardDenoms
 }
 
-// TransferRewardsToFeeCollector transfers all consumer rewards to the fee collector address
-func (k Keeper) TransferRewardsToFeeCollector(ctx sdk.Context) {
-	// 1. Get the denom whitelist from the store
-	denoms := k.GetAllConsumerRewardDenoms(ctx)
+// AllocateTokens performs rewards distribution to the community pool and validators
+// based on the Partial Set Security distribution specification.
+func (k Keeper) AllocateTokens(ctx sdk.Context, totalPreviousPower int64, bondedVotes []abci.VoteInfo) {
+	// return if there is no coins in the consumer rewards pool
+	if k.GetConsumerRewardsPool(ctx).IsZero() {
+		return
+	}
+
+	// Iterate over all registered consumer chains
+	for _, consumer := range k.GetAllConsumerChains(ctx) {
+
+		// transfer the consumer rewards to the distribution module account
+		// note that the rewards transferred are only consumer whitelisted denoms
```

Comment:
Are they "whitelisted" because of the whitelisting that took part in IBC's middleware `OnRecvPacket`?

Reply:
That's correct.